airbyte-ci/connectors/pipelines/pipelines/models/contexts/pipeline_context.py

```diff
@@ -37,26 +39,65 @@ class PipelineContext:
     dockerd_service: Optional[Service]
     started_at: Optional[datetime]
     stopped_at: Optional[datetime]
+    _default_excluded_files: Optional[list[str]]
+    _faster_excluded_files: Optional[list[str]]
 
     secrets_to_mask: List[str]
 
     PRODUCTION = bool(os.environ.get("PRODUCTION", False))  # Set this to True to enable production mode (e.g. to send PR comments)
 
-    DEFAULT_EXCLUDED_FILES = (
-        [".git", "airbyte-ci/connectors/pipelines/*"]
-        + glob("**/build", recursive=True)
-        + glob("**/.venv", recursive=True)
-        + glob("**/secrets", recursive=True)
-        + glob("**/__pycache__", recursive=True)
-        + glob("**/*.egg-info", recursive=True)
-        + glob("**/.vscode", recursive=True)
-        + glob("**/.pytest_cache", recursive=True)
-        + glob("**/.eggs", recursive=True)
-        + glob("**/.mypy_cache", recursive=True)
-        + glob("**/.DS_Store", recursive=True)
-        + glob("**/airbyte_ci_logs", recursive=True)
-        + glob("**/.gradle", recursive=True)
-    )
+    def faster_get_default_excluded_files(self):
+        if not self._faster_excluded_files:
+            all_files = glob("**", recursive=True, include_hidden=True)
+            self._faster_excluded_files = [
+                file
+                for file in all_files
+                if file.find("/build/") > 0
+                or file.endswith("/build")
+                or file.find("/.venv/") > 0
+                or file.endswith("/.venv")
+                or file.find("/secrets/") > 0
+                or file.endswith("/secrets")
+                or file.find("/__pycache__/") > 0
+                or file.endswith("/__pycache__")
+                or file.find(".egg-info/") > 0
+                or file.endswith(".egg-info")
+                or file.find("/.vscode/") > 0
+                or file.endswith("/.vscode")
+                or file.find("/.pytest_cache/") > 0
+                or file.endswith("/.pytest_cache")
+                or file.find("/.eggs/") > 0
+                or file.endswith("/.eggs")
+                or file.find("/.mypy_cache/") > 0
+                or file.endswith("/.mypy_cache")
+                or file.find("/.DS_Store/") > 0
+                or file.endswith("/.DS_Store")
+                or file.find("/airbyte_ci_logs/") > 0
+                or file.endswith("/airbyte_ci_logs")
+                or file.find("/.gradle/") > 0
+                or file.endswith("/.gradle")
```

Comment:
can we dynamically generate these or statement from a list with wildwards which looks like `DEFAULT_EXCLUDED_FILES`?